src/h5web/vis-packs/core/gammaScale.ts

```diff
@@ -0,0 +1,151 @@
+import { scalePower } from '@visx/scale';
+import type { Domain, GammaScaleConfig } from './models';
+import { interpolateNumber, interpolateRound } from 'd3-interpolate';
+import type {
+  InterpolatorFactory,
+  NumberValue,
+  ScalePower,
+  UnknownReturnType,
+} from 'd3-scale';
+import { isNumber } from 'lodash';
+
+function normalize(a: number, b: number) {
+  const extent = b - a;
+  return (x: number) => (extent !== 0 ? (x - a) / extent : 0.5);
+}
+
+type Range = number;
+type Output = Range;
+type Unknown = never;
+export type GammaScale = ScalePower<Range, Output, Unknown>;
+
+export function gammaScale(config?: GammaScaleConfig): GammaScale {
+  let _domain: number[] = config?.domain || [0, 1];
+  let _range: Range[] = config?.range || [0, 1];
+  let _exponent = config?.exponent ?? 1;
+  let _clamp = config?.clamp || false;
+  let _interpolate: InterpolatorFactory<Range, Output> = interpolateNumber;
+  let _unknown: Unknown;
+
+  const scaleFn = function (val: NumberValue) {
+    const [d0, d1] = _domain;
+    const [r0, r1] = _range;
+    const x = isNumber(val) ? val : val.valueOf();
+
```

Comment:
You could add a `NaN` check and early-return `_unknown` here, to make use of it.